src/main/java/org/piwik/java/tracking/CustomVariableList.java

```diff
@@ -6,20 +6,19 @@
  */
 package org.piwik.java.tracking;
 
-import java.util.HashMap;
-import java.util.Iterator;
-import java.util.Map;
-import java.util.Map.Entry;
 import javax.json.Json;
 import javax.json.JsonArrayBuilder;
 import javax.json.JsonObjectBuilder;
+import java.util.Iterator;
+import java.util.Map.Entry;
+import java.util.concurrent.ConcurrentHashMap;
 
 /**
  *
  * @author brettcsorba
  */
 class CustomVariableList{
-    private final Map<Integer, CustomVariable> map = new HashMap<>();
+    private final ConcurrentHashMap<Integer, CustomVariable> map = new ConcurrentHashMap<>();
```

Comment:
There's not a need to have calls to the map be thread safe, so we inherit unnecessary overhead using a concurrent hash map.  If we want to remove the call to putIfAbsent, we can refactor that logic to use containsKey/put in the while loop.